### PR TITLE
support for intfloat/multilingual-e5-small request

### PR DESCRIPTION
**Pull request by ashiskumarnaik**
support for intfloat/multilingual-e5-small request
This PR resolves issue #123

### CODE REVIEW

**Comment by NirantK** [comment]
Failing tests. Please verify that the ONNX values do map to the Torch implementation first? You can try with some sample text and compare.

**Comment by ashiskumarnaik** [comment]
updated

File: fastembed/text/onnx_embedding.py [hunk 1]
```diff
@@ -130,6 +130,15 @@
...
     },
+    {
+        "model": "intfloat/multilingual-e5-small",
+        "dim": 384,
+        "description": "Multilingual model, e5-small. Recommend using this model for non-English languages",
+        "size_in_GB": 0.5,
+        "sources": {
+            "hf": "intfloat/multilingual-e5-small",
+        },
+    },
```
Comment (NirantK) [inline fastembed/text/onnx_embedding.py:141]: Should be added to the multilingual_e5 file perhaps?

File: tests/test_text_onnx_embeddings.py [hunk 2]
```diff
@@ -16,6 +16,7 @@
     "sentence-transformers/all-MiniLM-L6-v2": np.array([0.0259, 0.0058, 0.0114, 0.0380, -0.0233]),
     "sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2": np.array([0.0094, 0.0184, 0.0328, 0.0072, -0.0351]),
     "intfloat/multilingual-e5-large": np.array([0.0098, 0.0045, 0.0066, -0.0354, 0.0070]),
+    "intfloat/multilingual-e5-small": np.array([0.04931236, 0.02415175, -0.0384715, -0.08884481, 0.08710264]),
```
Comment (NirantK) [inline tests/test_text_onnx_embeddings.py:19]: This is not the vector you get after normalization: https://colab.research.google.com/drive/1tNdV3DsiwsJzu2AXnUnoeF5av1Hp8HF1?usp=sharing

This looks like you've copied the vector from the ONNX test case and pasted here, please don't do that!

**Review by NirantK — CHANGES_REQUESTED** [review]

**Comment by ashiskumarnaik** [comment]
Screenshot 2024-03-16 at 8 27 12 PM
Hi I am adding this vector values , I ran the test with this first five values and tests are passed.
Screenshot 2024-03-16 at 11 00 51 PM
vector values: [0.0493123643,  0.0241517499, -0.0384715013, -0.0888448060, 0.087102644]

File: tests/test_text_onnx_embeddings.py (hunk 2 above)
Comment (NirantK) [reply tests/test_text_onnx_embeddings.py]: This is NTO Resolved. The test vector needs to come from a source implementation which we can trust. For instance, I've used Sentence Transformers — which takes care of the proper normalization as recommended by the model creators. 

You've used the output from your own FastEmbed implementation — that's not the point of the test? How do you know that your implementation is correct — you don't. 

In fact, the Colab notebook I've shared proved that this implementation is wrong.

File: fastembed/text/onnx_embedding.py [hunk 3]
```diff
@@ -130,6 +130,15 @@
             "hf": "qdrant/gte-large-onnx",
         },
     },
+    {
```
Comment (NirantK) [inline fastembed/text/onnx_embedding.py:133]: Move this to the e5_onnx file please?

File: tests/test_text_onnx_embeddings.py [hunk 4]
```diff
@@ -16,6 +16,7 @@
     "sentence-transformers/all-MiniLM-L6-v2": np.array([0.0259, 0.0058, 0.0114, 0.0380, -0.0233]),
     "sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2": np.array([0.0094, 0.0184, 0.0328, 0.0072, -0.0351]),
     "intfloat/multilingual-e5-large": np.array([0.0098, 0.0045, 0.0066, -0.0354, 0.0070]),
+    "intfloat/multilingual-e5-small": np.array([0.0493123643, 0.0241517499, -0.0384715013, -0.0888448060, 0.087102644]),
```
Comment (NirantK) [inline tests/test_text_onnx_embeddings.py:19]: Please see previous review comments

**Review by NirantK — CHANGES_REQUESTED** [review]

**Comment by NirantK** [comment]
Hello! 

Thanks for taking the time for raising this PR. We'll try to support the model in case this PR doesn't go through.

In the meanwhile, I'm closing this pull request for now because of inactivity. Please feel free to open it once you've resolved conflicts and made the requested changes.

**Pull request closed without merging**